### PR TITLE
DOC: add section on setting random number seeds

### PR DESCRIPTION
**Pull request by tacaswell**
Document using 19680801 as the preferred random seed.

### CODE REVIEW

**Comment by afvincent** [comment]
Minor remark FWIW, shouldn´t we promote fixing the PRNG seed in examples as well? I think that @NelleV (or maybe @choldgraf, I don´t remember clearly 🐑) did some work a while ago to ensure that 19680801 was used as the seed for the PRNGs in *every* files where a call to a random function was made (not only the test-related ones). Do we want to make this clear in the docs too?

**Comment by tacaswell** [comment]
I think the samples are uniformly change but most of the tests are not (because it would require re-generating test images).  I decided against mentioning the examples here as this is a section on tests.

File: doc/devel/testing.rst [hunk 1]
```diff
@@ -139,6 +139,23 @@ execution (such as created figures or modified rc params). The pytest fixture
 :func:`~matplotlib.testing.conftest.mpl_test_settings` will automatically clean
 these up; there is no need to do anything further.
 
+Random data in tests
+--------------------
+
+Random data can is a very convenient way to generate data for examples,
+however the randomness is problematic for testing (as the tests
+must be deterministic!).  To work around this set the seed in the test.
```
Comment (afvincent) [inline doc/devel/testing.rst]: "in the test*s* ?": typo or correct English.

**Comment by afvincent** [comment]
Fair enough about restricting the mention to the tests in a doc section about testing.

File: doc/devel/testing.rst (hunk 1 above)
Comment (tacaswell) [reply doc/devel/testing.rst]: tweaked a bit.  The seed needs to be set in each test, just just once.

**Pull request merged**